### ISSUE TITLE
Autocomplete-python v1.17/0 not working

### ISSUE THREAD

**Issue by hallymer** [autocomplete-python/autocomplete-python#450]
After updating to **autocomplete-python v1.17.0**, the function description does not work.

So I downgrade to **autocomplete-python v1.16.0** and run it, but still, the function description does not work.

I looked for a reason not to work, "grammar2.7.txt" ~ "grammar3.7.txt" were in the parser file.

After duplicating on grammar3.x file, I changed it according to the Python version. And then restarted Atom.

The function description started to work!!

-----------------------------------------------------------------------------------------

### OS is based on Windows 10 and Python version is version 3.8
1. (https://github.com/autocomplete-python/autocomplete-python/releases) **Download version 1.16.0 from the link next to it.**

2. Unzip it once you have downloaded either "zip" or "tar.gz".

3. When unpacked, there are "keymaps" to "README.md" in the "autocomplete-python-1.16.0" folder, copy it(Ctrl+C).

4. Go to **C:\Users\UserID\\.atom\packages\autocomplete-python** and paste(Ctrl+V) the copied file.

5. Go to **C:\Users\UserID\\.atom\packages\autocomplete-python\lib\jedi\parser**

6. You will see the file **"grammar3.X.txt"**. Duplicate the file.

7. **Change the file name according to your version of Python.** (My python version is 3.8, so I have to write it as garmmar3.8.txt.)

8. When you're done, **Restart Atom.**
-----------------------------------------------------------------------------------------
This method is temporary until the next update of version 1.17.0

**I hope it helps a lot.** :smiley:

**Comment by hallymer** [comment]
Also, I found **second solution.**
1. Install a recent version of python and then checking the box to ** (very important) Add Python to PATH** during the installation. 
**OR** Manually add Python to Windows Path **(Google for more details:wink:)**
2.  Use command prompt(CMD) and then Verify jedi installation through **"pip install jedi"** command
3. **Reinstalling autocomplete-python v1.17.0**
--------------------------------------------------------------
### As a second solution, you can use autocomplete-python v1.17.0 right away without downgrades.

**Comment by iconberg** [comment]
Have the same issue on linux after updating. How to fix this on linux?

**Comment by hallymer** [comment]
Maybe In Linux, I think you can do sudo pip install jedi

**Comment by paul-grassl** [comment]
Somehow autocomplete-python can't find my jedi installation. If I start to type something in a python script, i get:
```
Traceback (most recent call last):
  File "/Users/paulgrassl/.atom/packages/autocomplete-python/lib/completion.py", line 8, in <module>
```
and also:
```
    import jedi
ModuleNotFoundError: No module named 'jedi'
```

However, if I do a `pip list` in my venv I see the latest jedi version...

is this also related to this? any advice is appreciated. (I already tried to uninstall jedi and autocomplete-python several times but this didn't change anything)

**Comment by hallymer** [comment]
@paul-grassl 
Have you tried **reinstalling autocomplete-python v1.17.0?**

**Comment by paul-grassl** [comment]
> @paul-grassl
> Have you tried **reinstalling autocomplete-python v1.17.0?**

yes I did. can it have something to do with the python path? ... but executing python commands works

**Comment by hallymer** [comment]
**It is related to the Python path.**
Did you check Path when installing Python?

If you did not check PATH when installing Python, 
refer to the link below and add the Python path.
[https://datatofish.com/add-python-to-windows-path/](https://datatofish.com/add-python-to-windows-path/)

**Comment by paul-grassl** [comment]
I'm on MacOS. And the path is fine imo (it's in my zshrc script). If it would be a python path problem then I wouldn't be able to execute Python commands right? The error that I'm getting is about not finding jedi.
Do you work with autocomplete-python version 1.16.0 or 1.17.0?

**Comment by hallymer** [comment]
What is the version of Python installed on the Mac?
**Python installed on Mac is set to 2.x version, that is, the default version.**
**So it may not work**

Please refer to the link below
[https://dev.to/malwarebo/how-to-set-python3-as-a-default-python-version-on-mac-4jjf](https://dev.to/malwarebo/how-to-set-python3-as-a-default-python-version-on-mac-4jjf)

**Comment by paul-grassl** [comment]
I have an alias in my zshrc script which makes python3 the default.

Thanks for your suggestions though, really appreciate it. I will also experiment again with maybe different venv etc.

**Comment by hallymer** [comment]
you changed the python version and it works fine?

**Comment by paul-grassl** [comment]
No, sorry maybe that was not clear. I just wanted to say that python 3 is my default, so it shouldn't be problem of python 2. Additionally I just wanted to thank you for your suggestions, however, it may be impossible for you to figure out this problem remotely and the only way is that I do trial and error :S

**Comment by hallymer** [comment]
Try with this command.

- **sudo apt-get install vim-python-jedi**
- **vim-addons install python-jedi**

Try the above command and tell me the result 😄

**Comment by hallymer** [comment]
This link will also be helpful.
[https://github.com/clduan/jedi-in-mac](https://github.com/clduan/jedi-in-mac)

**Comment by iconberg** [comment]
For me
    
    sudo pip install jedi
    sudo apt-get install vim-python-jedi
    sudo vim-addons install python-jedi

and reinstall atom 1.54 did not solve the problem, still no autocomplete.

**Comment by j2bbayle** [comment]
This is such an annoying bug… can't wait for a proper fix 🙌!
